### DISCUSSION TITLE
How does Snipe-IT implement OpenAPI?

### DISCUSSION THREAD

**Discussion by themegabyte** [snipe/snipe-it#11795] [Q&A]
Hello!

I just deployed snipe-it for my company and I noticed the amazing API documentation.

1) Can anyone point me in the right direction as to how this is generated? Please excuse my lack of knowledge. I tried looking but I couldn't find other then the comments in the source code.

I am documenting another API for a personal application and I wanted to use openAPI for that. I am currently writing it in YAML files as I don't like to include it in my javascript code (my app is in Javascript!). 

2) Does this happen "automagically" for SnipeIT? Or the SnipeIT API generation source code is not public.

Thank you!

**Comment by ghuysmans** [comment]
[This answer](https://github.com/snipe/snipe-it/issues/10328#issuecomment-973220273) might help you ;)
